core/src/test/java/com/es/core/dao/impl/JdbcPhoneDaoIntTest.java

```diff
@@ -0,0 +1,111 @@
+package com.es.core.dao.impl;
+
+import com.es.core.model.phone.Phone;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.jdbc.JdbcTestUtils;
+
+import javax.annotation.Resource;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@ContextConfiguration("classpath:/context/testApplicationContext-core.xml")
+@RunWith(SpringJUnit4ClassRunner.class)
+public class JdbcPhoneDaoIntTest {
+    @Resource
+    private JdbcTemplate jdbcTemplate;
+    @Resource
+    private JdbcPhoneDao jdbcPhoneDao;
+
+    private static final Phone phone1 = new Phone();
+    private static final Phone phone2 = new Phone();
+
+    private static void setupPhone(Phone phone, Long id, String brand, String model) {
+        phone.setId(id);
+        phone.setBrand(brand);
+        phone.setModel(model);
+    }
+
+    @BeforeClass
+    public static void setUpClass() {
+        setupPhone(phone1, 101L, "Xiaomi", "Mi 6");
+        setupPhone(phone2, 102L, "Samsung", "S 20");
+    }
+
+    @Before
+    public void setUp() {
+        JdbcTestUtils.deleteFromTables(jdbcTemplate, "phones");
+        jdbcPhoneDao.save(phone1);
```

Comment:
Hint: instead of it you can use some test demodata script to insert test values. Looks more convenient and it will increase performance in case of large project.